src/decorator-annotator.ts

```diff
@@ -53,7 +53,8 @@ class ClassRewriter extends Rewriter {
   }
 
   private decoratorsToLower(n: ts.Node): ts.Decorator[] {
-    return (n.decorators as ts.NodeArray<ts.Decorator>|| []).filter(d => this.shouldLower(d));
+    let decorators = (n.decorators || []) as ts.NodeArray<ts.Decorator>;
```

Comment:
Won't work because actual type of `n.decorators || []` is `NodeArray<ts.Decorator> | never[]`, i.e. we are actually performing a downcast here not upcast.

IMHO, the most readable pattern and one that works with and without SNC would be:
```ts
  if (n.decorators) {
    return n.decorators.filter(() => true);
  }
  return [];
```

Also, on to check this as is to match the change in g3 and we can change it later.